File: internal/extract/testdata/go/basic.go
package billing

import "fmt"

const Version = "1.0"

const (
	MaxRetries = 3
	MinDelay   = 100
)

type Money struct {
	Amount   int
	Currency string
}

type Processor interface {
	Process(m Money) error
}

type Handler = func(Money) error

type Amount int

func (m Money) Format() string {
	return fmt.Sprintf("%d %s", m.Amount, m.Currency)
}

func (m *Money) Add(other Money) {
	m.Amount += other.Amount
}

func Process(m Money) error {
	return nil
}

File: internal/extract/testdata/rust/basic.rs
const MAX_SIZE: u32 = 100;
static VERSION: &str = "1.0";

pub struct Money {
    pub amount: i64,
    currency: String,
}

pub enum Status {
    Active,
    Inactive,
}

pub trait Formatter {
    fn format(&self) -> String;
}

pub type Name = String;

impl Money {
    pub fn new(amount: i64) -> Self {
        Self { amount, currency: String::from("USD") }
    }

    pub fn display(&self) -> String {
        format!("{}", self.amount)
    }
}

impl Formatter for Money {
    fn format(&self) -> String {
        self.display()
    }
}

pub fn process(m: Money) {
    let _ = m;
}
